### PR TITLE
Use !Send types for wasm32 support.

### PR DESCRIPTION
**Pull request by tmpfs**
This depends upon landing wasm32 support for `bytes_stream()` in `reqwest`; I have opened a PR that adds support [here](https://github.com/seanmonstar/reqwest/pull/1576).

If we can land that PR and this one then this library should work in wasm32 😁

### CODE REVIEW

**Comment by EstebanBorai** [comment]
Hi! Any news on this? I'm relying on this crate for a project and interested on using support for wasm

**Comment by nab-os** [comment]
Hello @tmpfs 
I'm hitting a compilation error with your branch:
```
error[E0599]: no method named `bytes_stream` found for struct `Response` in the current scope
   --> reqwest-eventsource/src/event_source.rs:163:30
    |
163 |         let mut stream = res.bytes_stream().eventsource();
    |                              ^^^^^^^^^^^^ method not found in `Response`
```
I don't have it while compiling on the default x86 target.
Do you know where it comes from and how I could fix it ?
I am not used to wasm compilation

File: Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ categories = ["web-programming::http-client", "no-std", "parsing", "asynchronous
 
 [dependencies]
 eventsource-stream = "0.2.3"
-reqwest = { version = "0.11", default-features = false, features = ["stream"] }
+reqwest = { version = "0.11", path = "../reqwest", default-features = false, features = ["stream"] }
```
Comment (nab-os) [inline Cargo.toml:18]: Isn't this a local modification for development ?

Comment (FirelightFlagboy) [inline Cargo.toml:18]: Is the `path` value expected ?

`cargo check` fail (see below for the error message) when until I remove that field.

```txt
    Updating git repository `https://github.com/tmpfs/reqwest-eventsource.git`
error: no matching package named `reqwest` found
location searched: https://github.com/tmpfs/reqwest-eventsource.git?rev=6ce0923ccc7cae4ce39a600b35425a62edc632b5#6ce0923c
required by package `reqwest-eventsource v0.4.0 (https://github.com/tmpfs/reqwest-eventsource.git?rev=6ce0923ccc7cae4ce39a600b35425a62edc632b5#6ce0923c)`
    ... which satisfies git dependency `reqwest-eventsource` of package `sse-web v0.1.0 (/home/[Redacted]/Documents/platform_sse/sse-web)`
```

**Comment by tmpfs** [comment]
Hi @FirelightFlagboy, I don't need this anymore so won't continue to work on this, however, I noticed that `bytes_stream()` has been added to `reqwest` (see the thread in the linked issue), so it should be just a case of upgrading `reqwest` and merging this.

**Comment by jpopesculian** [comment]
closed with #13 thanks!!!

**Pull request closed without merging**